plugins/webp-uploads/picture-element.php

```diff
@@ -0,0 +1,154 @@
+<?php
+/**
+ * Add Picture Element support
+ * Description: Use <picture> element when image has more than one mime type.
+ * Experimental: No
+ *
+ * @package performance-lab
+ * @since n.e.x.t
+ */
+
+/**
+ * Potentially wrap an image tag in a picture element.
+ *
+ * @since n.e.x.t
+ *
+ * @param string $image         The image tag.
+ * @param string $context       The context for the image tag.
+ * @param int    $attachment_id The attachment id.
+ *
+ * @return string The new image tag.
+ */
+function webp_uploads_wrap_image_in_picture( string $image, string $context, int $attachment_id ): string {
+	if ( 'the_content' !== $context || ! webp_uploads_picture_element_enabled() ) {
+		return $image;
+	}
+	$image_meta              = wp_get_attachment_metadata( $attachment_id );
+	$original_file_mime_type = get_post_mime_type( $attachment_id );
+	if ( false === $original_file_mime_type || ! isset( $image_meta['sizes'] ) ) {
+		return $image;
+	}
+
+	// Collect all of the sub size image mime types.
+	$mime_type_data = array();
+	foreach ( $image_meta['sizes'] as $size ) {
+		if ( isset( $size['sources'] ) && isset( $size['width'] ) && isset( $size['height'] ) ) {
+			foreach ( $size['sources'] as $mime_type => $data ) {
+				$mime_type_data[ $mime_type ]                         = isset( $mime_type_data[ $mime_type ] ) ? $mime_type_data[ $mime_type ] : array();
+				$mime_type_data[ $mime_type ]['w'][ $size['width'] ]  = $data;
+				$mime_type_data[ $mime_type ]['h'][ $size['height'] ] = $data;
+			}
+		}
+	}
+	$sub_size_mime_types = array_keys( $mime_type_data );
+
+	/**
+	 * Filter the image mime types that can be used for the <picture> element.
+	 *
+	 * Default is: ['image/webp', 'image/jpeg']. Returning an empty array will skip using the picture element.
+	 *
+	 * The mime types will output in the picture element in the order they are provided.
+	 * The original image will be used as the fallback image for browsers that don't support the picture element.
+	 *
+	 * @since n.e.x.t
+	 * @param array<string> $mime_types    Mime types than can be used.
+	 * @param int           $attachment_id The id of the image being evaluated.
+	 */
+	$enabled_mime_types = apply_filters(
+		'webp_uploads_picture_element_mime_types',
+		array(
+			'image/webp',
+			'image/jpeg',
+		),
+		$attachment_id
+	);
+
+	$mime_types = array_filter(
+		$enabled_mime_types,
+		static function ( $mime_type ) use ( $sub_size_mime_types ) {
+			return in_array( $mime_type, $sub_size_mime_types, true );
+		}
+	);
+
+	// No eligible mime types.
+	if ( ! $mime_types ) {
```

Comment:
PHPStan is complaining for me here:

> phpstan: Only booleans are allowed in a negated boolean, array<string> given.

This would also be needed with #1241. 

```suggestion
	if ( count( $mime_types ) === 0 ) {
```